project/MiMa.scala

```diff
@@ -3,19 +3,55 @@
  */
 package akka
 
-import com.typesafe.tools.mima.plugin.MimaKeys.{bbcIssueFilters, previousArtifacts}
-import com.typesafe.tools.mima.plugin.MimaPlugin.mimaDefaultSettings
 import sbt._
+import sbt.Keys._
+import com.typesafe.tools.mima.plugin.MimaPlugin
+import com.typesafe.tools.mima.plugin.MimaPlugin.autoImport._
 
 object MiMa extends AutoPlugin {
 
+  override def requires = MimaPlugin
   override def trigger = allRequirements
 
-  override val projectSettings = mimaDefaultSettings ++ Seq(
-    previousArtifacts := Set.empty,
-    bbcIssueFilters ++= mimaIgnoredProblems
+  override val projectSettings = Seq(
+    mimaBackwardIssueFilters ++= mimaIgnoredProblems,
+    mimaPreviousArtifacts := akkaPreviousArtifacts(name.value, organization.value, scalaBinaryVersion.value)
   )
 
+  def akkaPreviousArtifacts(projectName: String, organization: String, scalaBinaryVersion: String): Set[sbt.ModuleID] = {
+    val versions = {
+      val akka23Versions = Seq("2.3.11", "2.3.12", "2.3.13", "2.3.14")
+      val akka24NoStreamVersions = Seq("2.4.0", "2.4.1")
+      val akka24StreamVersions = Seq("2.4.2")
+      val akka24NewArtifacts = Seq(
+        "akka-cluster-sharding",
+        "akka-cluster-tools",
+        "akka-cluster-metrics",
+        "akka-persistence",
+        "akka-distributed-data-experimental",
+        "akka-persistence-query-experimental"
+      )
+      val akka242NewArtifacts = Seq(
+        "akka-stream",
+        "akka-stream-testkit",
+        "akka-http-core",
+        "akka-http-experimental",
+        "akka-http-testkit",
+        "akka-http-jackson-experimental",
+        "akka-http-spray-json-experimental",
+        "akka-http-xml-experimental"
+      )
```

Comment:
Have we given up on BC for those? Otherwise it is good to check experimental also to not accidentilly change something.